src/main/java/org/jabref/logic/externalfiles/LinkedFileHandler.java

```diff
@@ -86,7 +86,9 @@ public boolean renameToName(String targetFileName, boolean overwriteExistingFile
         }
 
         final Path oldPath = oldFile.get();
-        final Path newPath = oldPath.resolveSibling(targetFileName);
+        final String oldExtension = FileUtil.getFileExtension(oldPath).orElse("");
+        final boolean hasExtension = FileUtil.getFileExtension(targetFileName).isPresent();
```

Comment:
Is this tested? Does `getFileExtension(...)` return null when there is no extension provided? Or an empty string?
Because `isPresent()` returns false only for null values.

Reply:
Also try out this case: what happens if renamed filename = `f1.` (trailing dot but no extension).

Reply:
I am currently writing tests for this functionality. Right now the result of renaming file: **f1.pdf** to f1. is **f1..pdf**, if this is not the expected result, please let me know what the result should be so I can implement tests correctly.

Reply:
Yes, this is alright!